public/app/features/dashboard/containers/DashboardPage.tsx

```diff
@@ -277,6 +280,27 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
       this.props.notifyApp(createErrorNotification(`Panel not found`));
       locationService.partial({ editPanel: null, viewPanel: null });
     }
+
+    if (config.featureToggles.nestedFolders) {
+      const { folderTitle, folderUid } = dashboard.meta;
+      if (
+        folderTitle &&
+        folderUid &&
+        this.state.pageNav &&
+        this.state.pageNav.parentItem?.text !== folderTitle &&
+        !this.state.isGettingFolderInfo
+      ) {
+        this.setState({ isGettingFolderInfo: true });
+        const { folderNav } = await loadFolderPage(folderUid);
+        this.setState({
+          pageNav: {
+            ...this.state.pageNav,
+            parentItem: folderNav,
+          },
+          isGettingFolderInfo: false,
+        });
+      }
+    }
   }
```

Comment:
I don't get this, this should all be done in the backend, when we load the dashboard from the API we should have all the information in the result from what folder and parent folder it's in

Reply:
Like the current api sets meta fields for folderTitle etc, we should not have to add extra loading steps here making the breadcrumbs flicker & load when loading something that is in a nested folder

Reply:
But if adding folder hierarchy in the backend is impossible for some reason, then at least we should handle this in 
https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/services/DashboardLoaderSrv.ts#L37   and not in the UI component .

Reply:
i agree 💯%... this was basically an attempt to get breadcrumbs working early for 9.5.0 without backend changes. but now we've missed 9.5.0, i think we can wait for the proper backend changes. otherwise we have this code which will always be terrible... we have to **sequentially** call the backend for folder details 👎

Reply:
Also agree 100% @torkelo.

It's on the agenda to discuss with backend 😌 

![image](https://user-images.githubusercontent.com/46142/229805413-a3616d08-56bf-433a-8c80-42905ccd6c8d.png)